### ISSUE TITLE
Better support for HEIC files

### ISSUE THREAD

**Issue by PoeHaH** [NicolasCARPi/dropzone#26]
Images taken with iPhone cameras are often stored in a raw format with ".heic" extension. Their mime type is "image/heic" so Dropzone tries to render a thumbnail preview, but it doesn't work: 
<img width="431" alt="image" src="https://github.com/user-attachments/assets/f7664824-ea3f-44c5-a87f-7d5e31379659">

In the source code, we can see why the thumbnail can't be displayed:
<img width="584" alt="image" src="https://github.com/user-attachments/assets/b1bdb580-7404-4c2a-a82e-f4b2a3d82630">

In short, the dataUrl is not a string, but an object.

I propose to fix this by making a simple change to the `thumbnail()` function to check if the `dataUrl` is a string. 
<img width="636" alt="image" src="https://github.com/user-attachments/assets/f5e2e2d5-cf7c-460a-8417-dbb4ac67cdbf">

Would you be open to add this into the next release? I

**Comment by NicolasCARPi** [comment]
You can start with this document: https://github.com/NicolasCARPi/dropzone/blob/master/CONTRIBUTING.md

After if you never opened a PR on GitHub, you can read about it in their doc: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request

**Comment by NicolasCARPi** [comment]
I think we can live fine without thumbnails for HEIC. Don't want to add a dependency.

What we could do instead, is provide an option to call a custom function for thumbnailing. This way, the end user could either use some JS libs to do it in the browser, or call a resizing api endpoint in their app.

**Comment by NicolasCARPi** [comment]
Yes, the current code needs fixing to handle unsupported images. We should allow-list the supported image formats, right now it's looking at  `image/*` mime, which was probably correct 10 years ago, but not anymore!
